lib/server.js

```diff
@@ -76,6 +76,7 @@ class Server extends KarmaEventEmitter {
       watcher: ['value', watcher],
       launcher: ['factory', Launcher.factory],
       config: ['value', config],
+      instantiatePlugin: ['factory', plugin.createInstantiatePlugin],
```

Comment:
Any particular reason why you want to expose it as a public API?

Reply:
#3633

Reply:
Ideally we would not put this in the injector but rather just import it into code in the karma core.  But it depends upon `emitter` which does not exist until `server` is created.  So `instantiatePlugin` would have to be created by eg `server` and exported in `server.js`.  Is that worth the effort?

Reply:
Maybe get rid of `createInstantiatePlugin` factory and pass `injector` to `instantiatePlugin` as a parameter. But if it is too inconvenient, I'm fine with the current code as well. After all, we don't really have a clear definition of a public API, so may as well just add a comment.

Reply:
Ok I dug into this some more.  

The di system provides two features: module separation for unit-testability and object publication for API.  Both or neither.  

We can install the entry when server boots or later, but to allow `preprocessor` to be unit tested independent of `plugin`, we need the later to be installed in the di table.